portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -58,17 +63,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String newComment = request.getParameter("new-comment");
     long timestamp = System.currentTimeMillis();
 
+    Document doc =
+        Document.newBuilder().setContent(newComment).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    float score = sentiment.getScore();
+    languageService.close();
+
     Entity commentEntity = new Entity("comment");
     commentEntity.setProperty("message", newComment);
     commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("sentiment", score);
     datastore.put(commentEntity);
 
     response.sendRedirect("/");
   }
 
-  private String toJson(ArrayList<String> list) {
+  private String toJson(ArrayList<Comment> list) {
       Gson gson = new Gson();
       String json = gson.toJson(list);
       return json;
   }
 }
+
+class Comment {
+    String comment;
+    Double score;
+    public Comment(String comment, Double score) {
+        this.comment = comment;
+        this.score = score;
+    }
+}
```

Comment:
Better for this to be in a separate file and add getters and setters (IDE can probably auto generate it for you). I know it adds a bunch of boilerplate, but it's better for a few reasons:
1. this class might be used in other places in the future, e.g. if your api gets big and gets split into multiple files, they would want to access the same Comment object
2. getters and setters allow adding validation before setting the value, e.g. checking score falls within a certain range

Reply:
That makes sense, thanks for the feedback!